pkg/handlers/v1/nexposeassetproducer.go

```diff
@@ -47,6 +49,8 @@ func (h *NexposeScannedAssetProducer) Handle(ctx context.Context, in ScanInfo) {
 						logger.Error(logs.ProducerFailure{
 							Reason: err.Error(),
 						})
+					} else {
+						totalAssetsProduced++
```

Comment:
task: This is a data race. Multiple background goroutines will be competing to write a new value for the variable. We run all of our tests with the `-race` flag which usually catches this condition. It makes me wonder if have insufficient test coverage for this code branch.

At the very least, you'll need to either add in a `sync.Mutex` to guard access to the counter variable or use the `atomic` increment functions (https://golang.org/pkg/sync/atomic/).

Reply:
but at the bottom of the function, don't we wait on all the wait groups -> wait until all the goroutines finish? I don't see why we need locks around totalAssetsProduced.

Reply:
The barrier guarantees that all of the background goroutines are exited but it does not guarantee ordering or concurrency of the routines. Since the Go runtime is multi-threaded it is possible, and likely, that multiple routines will be processing at the same time. Because of this it is possible that two routines will attempt to modify the value at the same time which can result in corruption because incrementing is a read-then-write operation.

Reply:
fixed!